src/DataStructures/ComplexModalVector.hpp

```diff
@@ -0,0 +1,134 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class Data.
+
+#pragma once
+
+#include <complex>
+
+#include "DataStructures/ModalVector.hpp"
+#include "DataStructures/VectorImpl.hpp"
+#include "Utilities/PointerVector.hpp"  // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+
+class ComplexDataVector;
+
+// IWYU pragma: no_include <blaze/math/expressions/DVecMapExpr.h>
+// IWYU pragma: no_include <blaze/math/typetraits/IsVector.h>
+// IWYU pragma: no_include "DataStructures/DataVector.hpp"
+// IWYU pragma: no_forward_declare ConstantExpressions_detail::pow
+
+/*!
+ * \ingroup DataStructuresGroup
+ * \brief A class for storing spectral coefficients on a mesh.
+ *
+ * \details A ComplexModalVector holds an complex array of spectral
+ * coefficients, and can be either owning (the array is deleted when the
+ * ModalVector goes out of scope) or non-owning, meaning it just has a pointer
+ * to an array.
+ *
+ * Only basic mathematical operations are supported with ComplexModalVectors. In
+ * addition to addition, subtraction, multiplication, division, there are the
```

Comment:
Whatever you decide to do for documenting the allowed operations in `ModalVector` please do the same here for consistency :)